File: internal/world/world.go
package world

import (
    "k-town/internal/config"
)

type LocationType string

const (
    Square     LocationType = "square"
    Workshop   LocationType = "workshop"
    Wilderness LocationType = "wilderness"
)

type Location struct {
    ID       string
    Name     string
    Type     LocationType
    Agents   []string
    Events   []string
    Properties map[string]interface{}
}

type World struct {
    Time      int64
    Weather   string
    Locations map[string]*Location
}

func New(cfg config.Config) *World {
    w := &World{
        Weather:   "clear",
        Locations: make(map[string]*Location),
    }
    w.Locations["square"] = &Location{ID: "square", Name: "Town Square", Type: Square}
    w.Locations["workshop"] = &Location{ID: "workshop", Name: "Workshop", Type: Workshop}
    w.Locations["wilderness"] = &Location{ID: "wilderness", Name: "Wilderness", Type: Wilderness}
    return w
}

func (w *World) Advance(tick int64) {
    w.Time = tick
    hour := tick % 24
    if hour == 6 {
        w.weatherChange()
    }
}

func (w *World) weatherChange() {
    weathers := []string{"clear", "cloudy", "rainy"}
    w.Weather = weathers[w.Time%3]
}

File: server/world.py
from .models import WorldState
from typing import Dict

class World:
    def __init__(self):
        self.state = WorldState()
        self.locations = {
            "square":{"id":"square","name":"Town Square","type":"square","agents":[]},
            "workshop":{"id":"workshop","name":"Workshop","type":"workshop","agents":[]},
            "wilderness":{"id":"wilderness","name":"Wilderness","type":"wilderness","agents":[]},
        }
        self.state.locations = self.locations
        self._weathers = ["clear","cloudy","rainy"]

    def advance(self, tick:int):
        self.state.tick = tick
        if tick % 24 == 6:
            self._change_weather()

    def _change_weather(self):
        self.state.weather = self._weathers[self.state.tick % 3]

    def add_agent_to_location(self, agent_id, location):
        if location in self.locations and agent_id not in self.locations[location]["agents"]:
            self.locations[location]["agents"].append(agent_id)

    def remove_agent_from_location(self, agent_id, location):
        if location in self.locations and agent_id in self.locations[location]["agents"]:
            self.locations[location]["agents"].remove(agent_id)

    def to_dict(self):
        return {"tick":self.state.tick,"weather":self.state.weather,
                "locations":{k:{**v,"agents":list(v["agents"])} for k,v in self.locations.items()}}
